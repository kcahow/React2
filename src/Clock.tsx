import React, { useState, useEffect } from "react";

export const Clock: React.FC = () => {
  const [time, setTime] = useState(new Date());

  useEffect(() => {
    const timerID = setInterval(() => setTime(new Date()), 1000);
    return () => clearInterval(timerID);
  }, []);

  return (
    <div>
      <div>Current Time</div>
      <p>{time.toLocaleTimeString()}</p>
    </div>
  );
};
