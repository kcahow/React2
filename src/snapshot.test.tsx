import React from "react";
import renderer from "react-test-renderer";
import { Clock } from "./Clock"; // Adjust the import path as necessary

test("Clock component snapshot", () => {
  const component = renderer.create(<Clock />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
