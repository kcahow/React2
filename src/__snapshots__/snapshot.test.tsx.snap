// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock component snapshot 1`] = `
<div>
  <div>
    Current Time
  </div>
  <p>
    12:37:24 PM
  </p>
</div>
`;
